docs/develop.md

```diff
@@ -0,0 +1,233 @@
+# Development
+
+There are many ways to intall a development environment for uData.
+To get up and running fast, we recommand to use virtualenv and Docker/Docker-Compose.
+
+## System dependencies
+
+The uData development environment requires the following system dependencies
+(on an Debian/APT based system):
+
+* build-essential
+* pkg-config
+* python-dev
+* python-virtualenv
+* libjpeg-dev
+* zlib1g-dev
+* libpng12-dev
+* libtiff5-dev
+* libfreetype6-dev
+* liblcms2-dev
+* libopenjpeg-dev
+* libwebp-dev
+* libpng12-dev
+* libxml2-dev
+* libxslt1-dev
+* liblzma-dev
+
+```shell
+apt-get install build-essential pkg-config python-dev python-virtualenv libjpeg-dev zlib1g-dev libpng12-dev libtiff5-dev libfreetype6-dev liblcms2-dev libopenjpeg-dev libwebp-dev libpng12-dev libxml2-dev libxslt1-dev liblzma-dev
+```
+
+## Python dependencies
+
+It is recommended to work within a virtualenv to ensure proper dependencies isolation.
+
+It is recommanded to have a parent workspace where project are cloned,
+and where data are stored...
+It will be designated by `$WORKSPACE` until the end of the documentation.
+
+```shell
+$ mkdir $WORKSPACE && cd $WORKSPACE
+```
+
+First, create and activate your virtualenv, with one of the following command
+(depending on you using virtualenv or virtualenvwrapper):
+
+```shell
+$ virtualenv . && source bin/activate  # for virtualenv
+$ mkvirtualenv udata && setvirtualenvproject .  # for virtualenvwrapper
+$ mkproject udata  # for virtualenvwrapper with configured project dir
+```
+
+uData expect to run with Python 2.7,
+so if your system is using Python 3 as primary Python installation,
+don't forget to specify the python version with the `-p` parameter:
+
+```shell
+$ virtualenv -p /bin/python2 .
+$ mkvirtualenv udata -p /bin/python2
+$ mkproject -p /bin/python2 udata
+```
+
+Clone the uData project into your workspace:
+
+```shell
+$ git clone https://github.com/etalab/udata.git
+```
+
+Install the development dependencies and install the project as editable:
+
+```shell
+$ pip install -r udata/requirements/develop.pip
+$ pip install --no-deps -e udata/
+```
+
+## Middleware installation
+
+You can use native middleware packages or docker images (prefered method)
+
+### The docker way
+
+This is the prefered method as it does not depends of version provided by your OS.
+You need to have [Docker][] installed and working for your user.
```

Comment:
No this this the syntax to reference a link with the same name (link references are case-insensitives).
I hesitated to remove the rule [`no-shortcut-reference-link`](https://github.com/wooorm/remark-lint/blob/master/doc/rules.md#no-shortcut-reference-link) which allow to simply write `[Docker]`.
What do you prefer ?